packages/iTwinUI-react/src/core/Dialog/DialogTitleBar.tsx

```diff
@@ -9,48 +9,78 @@ import { useTheme } from '@storybook/theming';
 import { IconButton } from '../Buttons';
 import '@itwin/itwinui-css/css/dialog.css';
 import { DialogContextProps, useDialogContext } from './DialogContext';
+import { DialogTitleBarTitle } from './DialogTitleBarTitle';
 
 export type DialogTitleBarProps = {
   /**
-   * Dialog title bar content.
+   * Dialog title bar content. If passed, then `title` prop is ignored.
    */
-  children: React.ReactNode;
+  children?: React.ReactNode;
+  /**
+   * Dialog title.
+   */
+  title?: React.ReactNode;
 } & Pick<DialogContextProps, 'isDismissible' | 'onClose'> &
-  React.ComponentPropsWithRef<'div'>;
+  Omit<React.ComponentPropsWithRef<'div'>, 'title'>;
 
 /**
  * Dialog title bar. Recommended to be used as a child of `Dialog`.
  * @example
- * <Dialog.TitleBar>My dialog title</Dialog.TitleBar>
+ * <Dialog.TitleBar title='My dialog title' />
+ * @example
+ * <Dialog.TitleBar>
+ *   <Dialog.TitleBar.Title>My dialog title</Dialog.TitleBar.Title>
+ *   <IconButton
+ *     size='small'
+ *     styleType='borderless'
+ *     onClick={onClose}
+ *     aria-label='Close'
+ *   >
+ *     <SvgClose />
+ *   </IconButton>
+ * </Dialog.TitleBar>
  */
-export const DialogTitleBar = React.forwardRef<
-  HTMLDivElement,
-  DialogTitleBarProps
->((props, ref) => {
-  const dialogContext = useDialogContext();
-  const {
-    children,
-    isDismissible = dialogContext.isDismissible,
-    onClose = dialogContext.onClose,
-    className,
-    ...rest
-  } = props;
-  useTheme();
-  return (
-    <div className={cx('iui-dialog-title-bar', className)} ref={ref} {...rest}>
-      <div className='iui-dialog-title'>{children}</div>
-      {isDismissible && (
-        <IconButton
-          size='small'
-          styleType='borderless'
-          onClick={onClose}
-          aria-label='Close'
-        >
-          <SvgClose />
-        </IconButton>
-      )}
-    </div>
-  );
-});
+export const DialogTitleBar = Object.assign(
+  React.forwardRef<HTMLDivElement, DialogTitleBarProps>((props, ref) => {
+    const dialogContext = useDialogContext();
+    const {
+      children,
+      title,
+      isDismissible = dialogContext.isDismissible,
+      onClose = dialogContext.onClose,
```

Comment:
Does it really need `onClose`? The Dialog should already have it.

I would say, if there is no context then don't even show the close icon.

Reply:
It is using it for the close icon click handler so I would say it is needed if someone for some reason decides to use it outside context.

Reply:
If someone uses it outside the context, then they should provide their own icon imo.

Reply:
idk, it doesn't hurt to have these props if they are supported trough context.